internal/shellgen/scripts.go

```diff
@@ -82,39 +81,53 @@ func WriteScriptsToFiles(devbox devboxer) error {
 	return nil
 }
 
-func WriteScriptFile(devbox devboxer, name string, body string) (err error) {
-	script, err := os.Create(ScriptPath(devbox.ProjectDir(), name))
+func writeHookFile(devbox devboxer, body string) (err error) {
+	script, err := createScriptFile(devbox, HooksFilename)
 	if err != nil {
 		return errors.WithStack(err)
 	}
-	defer func() {
-		cerr := script.Close()
-		if err == nil {
-			err = cerr
-		}
-	}()
-	err = script.Chmod(0755)
+	defer func() { _ = script.Close() }() // best effort: close file
+
+	_, err = script.WriteString(body)
+	return errors.WithStack(err)
+}
+
+func WriteScriptFile(devbox devboxer, name string, body string) (err error) {
+	script, err := createScriptFile(devbox, name)
 	if err != nil {
 		return errors.WithStack(err)
 	}
+	defer func() { _ = script.Close() }() // best effort: close file
 
 	if featureflag.ScriptExitOnError.Enabled() {
-		// Fish cannot run scripts with `set -e`.
-		// NOTE: Devbox scripts will run using `sh` for consistency. However,
-		// init_hooks in a fish shell will run using `fish` shell, and need this
-		// check.
-		isFish, err := devbox.IsUserShellFish()
-		if err != nil {
-			return errors.WithStack(err)
-		}
-		if !isFish {
-			body = fmt.Sprintf("set -e\n\n%s", body)
-		}
+		// NOTE: Devbox scripts will run using `sh` for consistency.
+		// However, we need to disable this for `fish` shell if/when we allow this for init_hooks,
+		// since init_hooks run in the host shell, and not `sh`.
```

Comment:
I don't think we'll never be able to add this to init hooks unless we decide not to source.